### ISSUE TITLE
Pulumi up does not respect aws:profile when secretsprovider is specified

### ISSUE THREAD

**Issue by michael-golden** [pulumi/pulumi#4663]
**Given:**
* The AWS cli is configured with multiple credential profiles;
* The stack is initialized with `pulumi init <stackName> --secrets-provider="awskms://<kms-id>?region=us-east-1`, 

**Result:** the `pulumi up` command fails with:
```
error: getting secrets manager: secrets (code=Unknown): AccessDeniedException: The ciphertext refers to a customer master key that does not exist, does not exist in this region, or you are not allowed to access.
        status code: 400, request id: xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
```

The aforementioned error is caused by the underlying GO-sdk not using the aws profile.

If you provide the the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` as environment variables prior to running `pulumi up`, then the command completes without issue.

It would be nice to have a --profile argument for stack initialization, so that the secrets provider uses the provided profile configuration; in addition, the `pulumi up` command should pass the aws credentials to the encryption secrets provider.

**Version:** v2.2.1

**Comment by lukehoban** [comment]
Thanks for reporting this @michael-golden.

The issue here is actually a bit broader - there is no currently no way to explicitly configure any secrets-provider explicitly, they must be configured via environment variables.  I've opened https://github.com/pulumi/pulumi/issues/4665 to track that broader issue, since it's come up a few times.

> does not respect `aws:profile`

This part is pretty much by-design, as that is a setting on the default provider, which may not be the same as the AWS account/profile you want to use for secrets storage.  I'm not certain we could even use that same configuration (the value of it might be encrypted by the encryption mechanism we are trying to initialize!), but as outlined in #4665, I also suspect we will need a way to solve for this more generally.

**Comment by lukehoban** [comment]
Actually going to close this one out as it will be addressed via #4665.
